Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/event/model/Certificate.java

```diff
@@ -141,13 +192,102 @@ public void setX509Cert(X509Certificate x509Cert) {
     }
 
     public String getValidStartDate() {
+        validStartDate = DateFormat.getDateTimeInstance(
+            DateFormat.LONG, DateFormat.LONG).format(x509Cert.getNotBefore());
         return validStartDate;
     }
 
     public void setValidStartDate(String validStartDate) {
         this.validStartDate = validStartDate;
     }
 
+    public String getValidEndDate() {
+        validEndDate = DateFormat.getDateTimeInstance(DateFormat.LONG, DateFormat.LONG).format(x509Cert.getNotAfter());
+        return validEndDate;
+    }
+
+    public void setValidEndDate(String validEndDate) {
+        this.validEndDate = validEndDate;
+    }
+
+    public String getIssuerName() {
+        issuerName = StringUtils.isEmpty(x509Cert.getIssuerDN().getName()) ? EMPTY_FIELD
```

Comment:
Do everything in the CertificateManagerServiceImpl . Use the certificate class just for sending to/from information on the View.